cmd/gateclient/client.go

```diff
@@ -294,20 +295,19 @@ func (m *GatewayClient) Authenticate() error {
 			go http.ListenAndServe(":8085", nil)
 			// Note: leaving server connection open for scope of request, will be reaped on exit.
 
-			verifierBytes := make([]byte, 5)
-			if _, err := rand.Read(verifierBytes); err != nil {
-				util.UI.Error("Could not generate random string for code_verifier")
+			verifier, err := generateCodeVerifier()
+			if err != nil {
 				return err
 			}
-			verifier := string(verifierBytes)
+
 			codeChallenge := oauth2.SetAuthURLParam("code_challenge", verifier)
 			codeVerifier := oauth2.SetAuthURLParam("code_verifier", verifier)
```

Comment:
Should we be doing a sha256 hash of this given that we're capable?  Per the RFC "If the client is capable of using "S256", it MUST use "S256", as "S256" is Mandatory To Implement (MTI) on the server."

Reply:
Makes sense.